File: python/hexagon.py
from figure import Figure
import math

class Hexagon(Figure):
    def __init__(self, color: str, side: float):
        super().__init__(color)
        self.side = side

    def get_name(self):
        return "Hexagon"

    def perimeter(self):
        return 6 * self.side

    def area(self):
        return (3 * math.sqrt(3) * (self.side ** 2)) / 2

File: java/figure/src/figure/Hexagon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package figure;

/**
 *
 * @author eyro1
 */
public class Hexagon extends Figure {
    private double side;

    public Hexagon(String color, double side) {
        super(color);
        this.side = side;
    }

    @Override
    public double perimeter() {
        return 6 * side;
    }

    @Override
    public double area() {
        return (3 * Math.sqrt(3) * Math.pow(side, 2)) / 2;
    }

    @Override
    public String getName() {
        return "Hexagon";
    }
}

